### PR TITLE
Flag deprecated mds commands and omit deprecated mon commands in help output

### PR DESCRIPTION
**Pull request by batrick**

### CODE REVIEW

**Comment by gregsfortytwo** [comment]
This looks good to me but we should get some eyes from other team members given the interface change.
@athanatos, @liewegas, @jecluis, @dmick, thoughts?

**Comment by batrick** [comment]
Unfortunately `get_command_descriptions` is used somehow to check command validity somewhere. So deprecated commands no longer work at all. I'm figuring out why that's happening.

**Comment by liewegas** [comment]
Reviewed-by:

**Comment by batrick** [comment]
@gregsfortytwo, what do you think now? Is it alright leaving flags as an int in the json?

**Comment by gregsfortytwo** [comment]
Seems fine with me. Dan or @jcsp probably have more useful thoughts on json formatting/transmission than me, though.

File: src/ceph.in [hunk 1]
```diff
@@ -337,6 +337,8 @@ def format_help(cmddict, partial=None):
 
         if not cmd['help']:
             continue
+        if cmd['flags'] is not None and (cmd['flags'] & 0x6):
```
Comment (batrick) [inline src/ceph.in]: Done.

**Comment by batrick** [comment]
I've also removed the deprecated commands from `ceph(1)` but left the legacy section in `doc/cephfs/administration.rst`.

**Comment by dmick** [comment]
Looks OK to me; because I'm that kind of person, I wouldn't mind having an option to ceph(1) to display obsolete/deprecated commands too, but I'm not gonna cry if I don't get it.

**Pull request merged**